src/cargo/ops/registry.rs

```diff
@@ -134,7 +134,13 @@ pub fn publish(ws: &Workspace<'_>, opts: &PublishOpts<'_>) -> CargoResult<()> {
         let reg_name = publish_registry
             .clone()
             .unwrap_or_else(|| CRATES_IO_REGISTRY.to_string());
-        if !allowed_registries.contains(&reg_name) {
+        if allowed_registries.is_empty() {
+            bail!(
+                "`{}` cannot be published.\n\
+                 `publish` is set to `false` or an empty list in Cargo.toml and prevents publishing.",
```

Comment:
Should we mention the full path `package.publish`? I am not sure…

```suggestion
                 `package.publish` is set to `false` or an empty list in Cargo.toml and prevents publishing.",
```

Reply:
Good idea! I searched in `tests` dir with `pckage.` and I found `workspace.package.description was not defined` and `unexpected end of input while parsing minor version number for key package.version`. So I think it makes sense to add it.

Reply:
Updated! 
Thanks for your review!  💚 💙 💜 💛 ❤️